airbyte-integrations/connectors/source-netsuite-odbc/source_netsuite_odbc/streams.py

```diff
@@ -0,0 +1,424 @@
+
+from airbyte_cdk.sources.streams import Stream, IncrementalMixin
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Union, Final
+from airbyte_cdk.models import AirbyteMessage, AirbyteStream, SyncMode
+from datetime import date
+from airbyte_cdk.models import (
+    AirbyteMessage,
+    AirbyteStateMessage,
+    Type,
+    AirbyteStateType,
+    AirbyteStreamState,
+    StreamDescriptor,
+    FailureType,
+)
+import pyodbc
+from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from dateutil import parser
+from .errors import NETSUITE_CONNECTION_EXPIRED_FAILURE
+from .odbc_utils import NetsuiteODBCCursorConstructor
+
+# A stream's read method can return one of the following types:
+# Mapping[str, Any]: The content of an AirbyteRecordMessage
+# AirbyteMessage: An AirbyteMessage. Could be of any type
+StreamData = Union[Mapping[str, Any], AirbyteMessage]
+
+NETSUITE_PAGINATION_INTERVAL: Final[int] = 10000
+EARLIEST_DATE: Final[str] = date(2020, 1, 1)
+YEARS_FORWARD_LOOKING: Final[int] = 1
+# Sometimes, system created accounts can have primary key values < 0.  To be safe, we choose
+# an arbitrary value of -10000 as the starting value for our primary key
+STARTING_PRIMARY_KEY_VALUE: Final[int] = -10000
+
+GENERATING_INCREMENTAL_QUERY_WITH_NO_INCREMENTAL_COLUMN_ERROR: Final[str] = "We should not be generating an incremental query unless an incremental column exists."
+
+
+class NetsuiteODBCStream(Stream):
+
+    #  The functions before the init call are all used to initialize the stream
+    def get_primary_key_from_airbyte_stream(self, stream: AirbyteStream):
+      key = stream.primary_key
```

Comment:
should this be `key = stream.source_defined_primary_key`?

primary_key is not an AirbyteStream field

Reply:
https://arc.net/l/quote/cnmzjaje ->  Confused since in the docs it says primary_key